pkg/controller.v1alpha3/suggestion/suggestion_controller.go

```diff
@@ -154,6 +154,9 @@ func (r *ReconcileSuggestion) Reconcile(request reconcile.Request) (reconcile.Re
 		}
 	}
 
+	if err := r.updateSpec(instance, oldS); err != nil {
```

Comment:
There is a chance of error here. What if updateSpec succeeds while updateStatus fails?

Reply:
Yeah, I think it is a huge problem. Should we place algorithm setting in status instead of spec?